substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -119,84 +419,1339 @@ protected int peekData(DatagramPacket p) throws IOException {
         throw VMError.unsupportedFeature("Unimplemented: java.net.PlainDatagramSocketImpl.peekData(DatagramPacket)");
     }
 
+    /* Do not re-format commented out code: @formatter:off */
+    //   760  /*
+    //   761   * Class:     java_net_PlainDatagramSocketImpl
+    //   762   * Method:    receive
+    //   763   * Signature: (Ljava/net/DatagramPacket;)V
+    //   764   */
+    //   765  JNIEXPORT void JNICALL
+    //   766  Java_java_net_PlainDatagramSocketImpl_receive0(JNIEnv *env, jobject this,
+    //   767                                                jobject packet)
     @Substitute
     @SuppressWarnings({"static-method", "unused"})
-    protected void receive0(DatagramPacket p) throws IOException {
-        throw VMError.unsupportedFeature("Unimplemented: java.net.PlainDatagramSocketImpl.receive0(DatagramPacket)");
+    protected void receive0(DatagramPacket packet) throws IOException {
+
+        //   769      char BUF[MAX_BUFFER_LEN];
+        CCharPointer BUF = StackValue.get(JavaNetNetUtilMD.MAX_BUFFER_LEN(), CCharPointer.class);
+        //   770      char *fullPacket = NULL;
+        CCharPointer fullPacket = WordFactory.nullPointer();
+        //   771      int mallocedPacket = JNI_FALSE;
+        boolean mallocedPacket = false;
+
+        //   772      jobject fdObj = (*env)->GetObjectField(env, this, pdsi_fdID);
+        FileDescriptor fdObj = Util_java_net_DatagramSocketImpl.as_Target_java_net_DatagramSocketImpl(this).fd;
+
+        //   773      jint timeout = (*env)->GetIntField(env, this, pdsi_timeoutID);
+        int timeout = this.timeout;
+
+        //   775      jbyteArray packetBuffer;
+        byte[] packetBuffer;
+        //   776      jint packetBufferOffset, packetBufferLen;
+
+        int packetBufferOffset, packetBufferLen;
+
+        //   778      int fd;
+        int fd;
+
+        //   780      int n;
+        int n;
+        //   781      SOCKADDR remote_addr;
+        Socket.sockaddr remote_addr = StackValue.get(Socket.sockaddr.class);
+        //   782      int len;
+        CIntPointer len_Pointer = StackValue.get(CIntPointer.class);
+
+        //   783      jboolean retry;
+        boolean retry;
+
+        //   784  #ifdef __linux__
+        if (IsDefined.__linux__()) {
+            //   785      jboolean connected = JNI_FALSE;
+            boolean connected = false;
+            //   786      jobject connectedAddress = NULL;
+            InetAddress connectedAddress = null;
+            //   787      jint connectedPort = 0;
+            int connectedPort = 0;
+            //   788      jlong prevTime = 0;
+            long prevTime = 0;
+            //   789  #endif
+        }
```

Comment:
Well, this is not going to do what you want.  You have declared 4 locals that go out of scope when you leave the `then` branch.  On the other hand, I do not see where these variables are used, anywhere in this method.  So maybe the best thing to do is leave the C code as `//` comments and add a Java comment `/* Unused. */`.